app/controllers/admin/users_controller.rb

```diff
@@ -51,11 +56,21 @@ def show
     private
 
     def user_params
-      params.require(:user).permit(:name, :email, :password, :password_confirmation)
+      params.require(:user).permit(:name, :email, :password, :password_confirmation, :role)
     end
 
     def find_user
       @user = User.find(params[:id])
     end
+
+    def redirect_when_not_admin
+      return if authorize_admin!
```

Comment:
- 應該不需要 `#authorize_admin!`
- `authorize_admin!` 設計不符合慣例：回傳 boolean 的 method 名稱結尾應該為 `?`
```suggestion
      return if current_user&.admin?
```